test/unit/test_diagram.py

```diff
@@ -45,6 +47,37 @@ def test_job_agg_data():
     assert job_agg.number_of_columns() == 6
 
 
+def test_seq():
+    """
+    function to test the Class JobData inside autosubmit/monitor/diagram.py
+    """
+    arr = diagram._seq(100, 2, 10)
+    length = 0
```

Comment:
You can drop the `length` variable and just use a list comprehension to convert the iterator into a list.